crates/turborepo-lib/src/config/turbo_config.rs

```diff
@@ -0,0 +1,592 @@
+const DEFAULT_LOGIN_URL: &str = "https://vercel.com";
+const DEFAULT_TIMEOUT: u64 = 20;
+
+macro_rules! create_builder {
+    ($func_name:ident, $property_name:ident, $type:ty) => {
+        pub fn $func_name(mut self, value: $type) -> Self {
+            self.override_config.$property_name = value;
+            self
+        }
+    };
+}
+
+#[derive(Serialize, Deserialize, Default, Debug, PartialEq, Eq, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct ConfigurationOptions {
+    #[serde(alias = "apiurl")]
+    #[serde(alias = "ApiUrl")]
+    #[serde(alias = "APIURL")]
+    pub(crate) api_url: Option<String>,
+    #[serde(alias = "loginurl")]
+    #[serde(alias = "LoginUrl")]
+    #[serde(alias = "LOGINURL")]
+    pub(crate) login_url: Option<String>,
+    #[serde(alias = "teamslug")]
+    #[serde(alias = "TeamSlug")]
+    #[serde(alias = "TEAMSLUG")]
+    pub(crate) team_slug: Option<String>,
+    #[serde(alias = "teamid")]
+    #[serde(alias = "TeamId")]
+    #[serde(alias = "TEAMID")]
+    pub(crate) team_id: Option<String>,
+    pub(crate) token: Option<String>,
+    pub(crate) signature: Option<bool>,
+    pub(crate) preflight: Option<bool>,
+    pub(crate) timeout: Option<u64>,
+    pub(crate) enabled: Option<bool>,
+}
+
+#[derive(Default)]
+pub struct TurborepoConfigBuilder {
+    repo_root: AbsoluteSystemPathBuf,
+    override_config: ConfigurationOptions,
+
+    #[cfg(test)]
+    global_config_path: Option<AbsoluteSystemPathBuf>,
+    #[cfg(test)]
+    environment: HashMap<OsString, OsString>,
+}
+
+// Getters
+impl ConfigurationOptions {
+    pub fn api_url(&self) -> &str {
+        self.api_url.as_deref().unwrap_or(DEFAULT_API_URL)
+    }
+
+    pub fn login_url(&self) -> &str {
+        self.login_url.as_deref().unwrap_or(DEFAULT_LOGIN_URL)
+    }
+
+    pub fn team_slug(&self) -> Option<&str> {
+        self.team_slug.as_deref()
+    }
+
+    pub fn team_id(&self) -> Option<&str> {
+        self.team_id.as_deref()
+    }
+
+    pub fn token(&self) -> Option<&str> {
+        self.token.as_deref()
+    }
+
+    pub fn signature(&self) -> bool {
+        self.signature.unwrap_or_default()
+    }
+
+    pub fn enabled(&self) -> bool {
+        self.enabled.unwrap_or(true)
+    }
+
+    pub fn preflight(&self) -> bool {
+        self.preflight.unwrap_or_default()
+    }
+
+    pub fn timeout(&self) -> u64 {
+        self.timeout.unwrap_or(DEFAULT_TIMEOUT)
+    }
+}
+
+trait ResolvedConfigurationOptions {
+    fn get_configuration_options(self) -> Result<ConfigurationOptions, ConfigError>;
+}
+
+impl ResolvedConfigurationOptions for PackageJson {
+    fn get_configuration_options(self) -> Result<ConfigurationOptions, ConfigError> {
+        match &self.legacy_turbo_config {
+            Some(legacy_turbo_config) => {
+                let synthetic_raw_turbo_json: RawTurboJSON =
+                    serde_json::from_value(legacy_turbo_config.clone())
+                        .map_err(|_| ConfigError::Anyhow(anyhow!("global_de")))?;
```

Comment:
I think this error could use some description.

Reply:
Just needed to be deleted. I'd threaded it in but missed deletion.